driver-core/src/main/com/mongodb/internal/session/ServerSessionPool.java

```diff
@@ -174,7 +174,8 @@ public BsonDocument getIdentifier() {
 
         @Override
         public long advanceTransactionNumber() {
-            return transactionNumber++;
+            transactionNumber++;
```

Comment:
This slight change was necessary to make retryable write after a commit work properly, as otherwise a retryable write would share the transaction number with a committed (but itself retryable by the application) transaction.

Reply:
Is that the same as `return ++transactionNumber;`?